src/mxf_helper/JPEGXSMXFDescriptorHelper.cpp

```diff
@@ -0,0 +1,363 @@
+
+	GenericPictureEssenceDescriptor *comp_desc = dynamic_cast<GenericPictureEssenceDescriptor*>(mFileDescriptor);
+	GenericPictureEssenceDescriptor *comp_desc_in = dynamic_cast<GenericPictureEssenceDescriptor*>(file_desc_in);
+
+	if (comp_desc_in->haveFrameLayout() && comp_desc_in->getFrameLayout() != MXF_FULL_FRAME)
+		BMX_EXCEPTION(("Only progressive video frame layout is currently supported"));
+
+	SET_PROPERTY(FrameLayout)
+	SET_PROPERTY(SignalStandard)
+	SET_PROPERTY(AspectRatio)
+	SET_PROPERTY(ActiveFormatDescriptor)
+	SET_PROPERTY(VideoLineMap)
+	if ((mFlavour & MXFDESC_IMF_FLAVOUR)) {
+		BMX_ASSERT(comp_desc->getFrameLayout() == MXF_FULL_FRAME);
+		comp_desc->setDisplayF2Offset(0);
+	}
+	else {
+		SET_PROPERTY(FieldDominance)
+		SET_PROPERTY(DisplayF2Offset)
+	}
+	SET_PROPERTY(CaptureGamma)
+	SET_PROPERTY(CodingEquations)
+	SET_PROPERTY(ColorPrimaries)
+
+	SET_PROPERTY(MasteringDisplayPrimaries)
+	SET_PROPERTY(MasteringDisplayWhitePointChromaticity)
+	SET_PROPERTY(MasteringDisplayMaximumLuminance)
+	SET_PROPERTY(MasteringDisplayMinimumLuminance)
+
+	SET_PROPERTY(ActiveWidth)
+	SET_PROPERTY(ActiveHeight)
+	SET_PROPERTY(ActiveXOffset)
+	SET_PROPERTY(ActiveYOffset)
+
+	SET_PROPERTY(AlternativeCenterCuts);
+
+	if (dynamic_cast<CDCIEssenceDescriptor*>(mFileDescriptor)) {
+		CDCIEssenceDescriptor *comp_desc = dynamic_cast<CDCIEssenceDescriptor*>(mFileDescriptor);
+		CDCIEssenceDescriptor *comp_desc_in = dynamic_cast<CDCIEssenceDescriptor*>(file_desc_in);
+
+		if (!comp_desc->haveColorSiting() && comp_desc_in->haveColorSiting())
+			SetColorSitingMod(comp_desc_in->getColorSiting());
+		SET_PROPERTY(BlackRefLevel)
+		SET_PROPERTY(WhiteReflevel)
+		SET_PROPERTY(ColorRange)
+	}
+
+	if (dynamic_cast<RGBAEssenceDescriptor*>(mFileDescriptor)) {
+		RGBAEssenceDescriptor *comp_desc = dynamic_cast<RGBAEssenceDescriptor*>(mFileDescriptor);
+		RGBAEssenceDescriptor *comp_desc_in = dynamic_cast<RGBAEssenceDescriptor*>(file_desc_in);
+
+		SET_PROPERTY(ComponentMaxRef)
+		SET_PROPERTY(ComponentMinRef)
+		SET_PROPERTY(ScanningDirection)
+	}
+}
+
+void JPEGXSMXFDescriptorHelper::UpdateFileDescriptor(JXSEssenceParser *essence_parser)
+{
+	
+	mJPEGXSSubDescriptor->setJPEGXSPpih(essence_parser->GetJPEGXSPpih());
+	mJPEGXSSubDescriptor->setJPEGXSPlev(essence_parser->GetJPEGXSPlev());
+	mJPEGXSSubDescriptor->setJPEGXSWf(essence_parser->GetJPEGXSWf());
+	mJPEGXSSubDescriptor->setJPEGXSHf(essence_parser->GetJPEGXSHf());
+	mJPEGXSSubDescriptor->setJPEGXSNc(essence_parser->GetJPEGXSNc());
+	mJPEGXSSubDescriptor->setJPEGXSHsl(essence_parser->GetJPEGXSHsl());
+
+	int length = essence_parser->GetJPEGXSComponentTable().GetSize();
+	std::vector<uint8_t> tmp(length);
+	ByteArray tmpArr = essence_parser->GetJPEGXSComponentTable();
+	unsigned char* data = (tmpArr).GetBytes();
+	for (int i = 0; i < length; i++)
+	{
+		tmp[i] = (*data);
+		data++;
+	}
+	
+	// for loop each element in the data pointer to the array
+	mJPEGXSSubDescriptor->setJPEGXSComponentTable(tmp);
```

Comment:
Could this just be something like this?
```
start = essence_parser->GetJPEGXSComponentTable().GetBytes();
end = start + essence_parser->GetJPEGXSComponentTable().GetSize());
SetJPEGXSComponentTable(vector<uint8_t>(start, end));
```
This will set the property in the sub-descriptor and set `mComponentTable` which `UpdateFileDescriptor` will use to set the pixel layout. See also the comment below about the pixel layout being overriden.